modules/api-gateway/outputs.tf

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+output "api" {
+  description = "API."
+  value       = google_api_gateway_api.api
+}
+
+output "api_config" {
+  description = "API configs."
+  value       = google_api_gateway_api_config.api_config
+}
+
+output "api_config_id" {
+  description = "The identifiers of the API configs."
+  value       = google_api_gateway_api_config.api_config.api_config_id
+}
+
+output "api_id" {
+  description = "API identifier."
+  value       = google_api_gateway_api.api.api_id
+}
+
+output "default_hostname" {
+  description = "The default host names of the API gateway."
+  value       = google_api_gateway_gateway.gateway.default_hostname
+}
+
+output "gateway" {
+  description = "API gateways."
+  value       = google_api_gateway_gateway.gateway
+}
+
+output "gateway_id" {
+  description = "The identifiers of the API gateways."
+  value       = google_api_gateway_gateway.gateway.gateway_id
+}
+
+output "service_account" {
+  description = "Service account resource."
+  value       = try(google_service_account.service_account[0], null)
+}
+
+output "service_account_email" {
+  description = "The service account for creating API configs."
+  value       = local.service_account_email
+}
+
+output "service_account_iam_email" {
+  description = "The service account for creating API configs."
+  value = join("", [
```

Comment:
this will be "serviceAccount:" if local.service_account_email is null. Perhaps 
```hcl
value = local.service_account_email == null ? null : "serviceAccount:${local.service_account_email}"
```
is better?

makes more sense